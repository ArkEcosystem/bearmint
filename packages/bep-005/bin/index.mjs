#!/usr/bin/env node
import { join } from 'node:path'

import { globby } from 'globby'
import meow from 'meow'

process.env.NODE_ENV = 'test'

const { flags } = meow({
	flags: {
		config: {
			type: 'string',
		},
	},
	importMeta: import.meta,
})

const { files } = (await import(join(process.cwd(), flags.config))).default
const tests = (await globby(files)).map((path) => join(process.cwd(), path))

for (const test of tests) {
	await import(test)
}
