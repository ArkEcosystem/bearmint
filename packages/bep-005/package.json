{
	"name": "@bearmint/bep-005",
	"version": "0.0.0",
	"private": true,
	"description": "This library is the official reference implementation for BEP-005 (https://bep.dev/005)",
	"keywords": [
		"bearmint",
		"tendermint",
		"blockchain",
		"framework",
		"modular"
	],
	"homepage": "https://github.com/bearmint/bearmint",
	"repository": {
		"type": "git",
		"url": "git+https://github.com/bearmint/bearmint.git"
	},
	"funding": "https://funding.bearmint.com/",
	"license": "MIT",
	"author": "Bearmint Mining Company <hello@bearmint.com> (https://bearmint.com/)",
	"type": "module",
	"exports": "./distribution/index.js",
	"types": "./distribution/index.d.ts",
	"bin": {
		"roadrunner": "./bin/index.mjs"
	},
	"files": [
		"bin",
		"distribution"
	],
	"scripts": {
		"audit": "pnpm audit --fix --prod",
		"build": "pnpm run clean && tsc",
		"build:watch": "pnpm run clean && tsc -w",
		"clean": "rm -rf distribution",
		"depcheck": "depcheck --ignores=@bearmint/bep-004,@bearmint/bep-005,@bearmint/bep-031,@bearmint/bep-039,globby,meow",
		"deps:major": "ncu -u --target=latest",
		"deps:minor": "ncu -u --target=minor",
		"deps:patch": "ncu -u --target=patch"
	},
	"dependencies": {
		"@bearmint/bep-013": "workspace:^",
		"@bearmint/bep-018": "workspace:^",
		"@bearmint/bep-116": "workspace:^",
		"@sinonjs/fake-timers": "9.1.2",
		"ansi-styles": "6.2.1",
		"concordance": "5.0.4",
		"globby": "13.1.2",
		"lodash.get": "4.4.2",
		"lodash.has": "4.5.2",
		"meow": "^11.0.0",
		"nock": "13.2.9",
		"pretty-ms": "8.0.0",
		"sinon": "14.0.1",
		"strip-ansi": "7.0.1",
		"vitest": "^0.24.5"
	},
	"devDependencies": {
		"@types/node": "18.11.9",
		"@types/sinon": "10.0.13",
		"@types/sinonjs__fake-timers": "8.1.2"
	},
	"engines": {
		"node": ">=18",
		"pnpm": ">=7.9.0"
	},
	"publishConfig": {
		"access": "restricted"
	}
}
