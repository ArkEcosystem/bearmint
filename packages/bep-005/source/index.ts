export * from './expect.js'
export * from './fake.js'
export * from './nock.js'
export * from './spy.js'
export * from './stub.js'
