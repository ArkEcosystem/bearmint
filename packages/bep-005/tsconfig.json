{
	"compilerOptions": {
		"typeRoots": ["node_modules/@types"],
		"outDir": "distribution"
	},
	"exclude": ["node_modules", "test/**/**.ts"],
	"extends": "../../config/tsconfig.json",
	"include": ["source/**/**.ts"]
}
