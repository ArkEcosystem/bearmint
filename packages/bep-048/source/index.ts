export * from './meter.js'
export * from './service-provider.js'
