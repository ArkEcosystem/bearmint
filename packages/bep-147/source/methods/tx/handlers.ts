import { jsonToBytes } from '@bearmint/bep-009'
import type { Cradle, QueryHandler, QueryHandlerOpts } from '@bearmint/bep-013'

export class ListTxHandlers implements QueryHandler {
	readonly #cradle: Cradle

	public constructor(cradle: Cradle) {
		this.#cradle = cradle
	}

	async isMatch({ path }: QueryHandlerOpts) {
		return path[0] === 'tx' && path[1] === 'handlers'
	}

	async execute() {
		const result = {}

		for (const [name, versions] of Object.entries(this.#cradle.TxManifestRegistry.all())) {
			result[name] = {}

			for (const [version, manifest] of Object.entries(versions)) {
				result[name][version] = {
					checksum: manifest.checksum(),
					name: manifest.name(),
					version: manifest.version(),
				}
			}
		}

		return {
			code: 0,
			value: jsonToBytes(result),
		}
	}
}
