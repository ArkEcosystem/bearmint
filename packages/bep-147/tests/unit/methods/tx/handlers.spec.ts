import { makeContainer } from '@bearmint/bep-014'
import { makeGasMeterFactory } from '@bearmint/bep-048'
import { expect, describe, it } from 'vitest'
import { extendExpect } from '@bearmint/bep-005'
extendExpect(expect)

import { ListTxHandlers } from '../../../../source/methods/tx/handlers.js'

describe('ListTxHandlers', () => {
	it('should determine if the handler is a match', async () => {
		expect(
			await new ListTxHandlers().isMatch({
				data: undefined,
				path: ['tx', 'handlers'],
			}),
		).toBeTrue()

		expect(
			await new ListTxHandlers().isMatch({
				data: undefined,
				path: [],
			}),
		).toBeFalse()
	})

	it('should get all tx handlers', async () => {
		expect(
			await new ListTxHandlers({
				Container: makeContainer(),
				GasMeterFactory: makeGasMeterFactory(),
				TxManifestRegistry: {
					all() {
						return {
							handler: {
								version: {
									checksum() {
										return 'checksum'
									},
									name() {
										return 'name'
									},
									version() {
										return 'version'
									},
								},
							},
						}
					},
				},
			}).execute(),
		).toMatchSnapshot()
	})
})
