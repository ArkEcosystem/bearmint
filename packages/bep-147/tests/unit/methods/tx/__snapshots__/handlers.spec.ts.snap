// Vitest Snapshot v1

exports[`ListTxHandlers > should get all tx handlers 1`] = `
{
  "code": 0,
  "value": Uint8Array [
    123,
    34,
    104,
    97,
    110,
    100,
    108,
    101,
    114,
    34,
    58,
    123,
    34,
    118,
    101,
    114,
    115,
    105,
    111,
    110,
    34,
    58,
    123,
    34,
    99,
    104,
    101,
    99,
    107,
    115,
    117,
    109,
    34,
    58,
    34,
    99,
    104,
    101,
    99,
    107,
    115,
    117,
    109,
    34,
    44,
    34,
    110,
    97,
    109,
    101,
    34,
    58,
    34,
    110,
    97,
    109,
    101,
    34,
    44,
    34,
    118,
    101,
    114,
    115,
    105,
    111,
    110,
    34,
    58,
    34,
    118,
    101,
    114,
    115,
    105,
    111,
    110,
    34,
    125,
    125,
    125,
  ],
}
`;
